Add a gdextension online documentation guide. (#9966)
godotengine/godot-docs: tutorials/scripting/gdextension/gdextension_docs_system.rst

        ``#ff00ff``, see :ref:`doc_bbcode_in_richtextlabel_hex_colors`).

    - ``[color={code/name}]{text}[/color]``

Publishing documentation online
-------------------------------

> You may want to publish an online reference for your gdextension, akin to this website. The most important step is to build ``.rst`` files from your ``.xml`` ones:

Reviewer: ```suggestion
You may want to publish an online reference for your GDExtension, akin to this website. The most important step is to build reStructuredText (``.rst``) files from your XML class reference:
```

.. code-block:: none

>     # You'll need to have python installed for this command to work.

Reviewer: ```suggestion
    # You'll need to have Python installed for this command to work.
```

    curl -sSL https://raw.githubusercontent.com/godotengine/godot/master/doc/tools/make_rst.py | python3 - -o "docs/classes" -l "en" doc_classes

> Your ``.rst`` files will now be available in ``docs/classes/``. From here, you can use any documentation builder to create a website from them. `godot-docs <https://hosted.weblate.org/projects/godot-engine/godot-docs>`_ uses `Sphinx <https://www.sphinx-doc.org/en/master/>`_. You can use the repository as a basis to build your own documentation system. This is where you'll have to get a bit creative, but here's a rough outline of the steps involved:

Reviewer: ```suggestion
Your ``.rst`` files will now be available in ``docs/classes/``. From here, you can use
any documentation builder that supports reStructuredText syntax to create a website from them.
`godot-docs <https://hosted.weblate.org/projects/godot-engine/godot-docs>`_ uses `Sphinx <https://www.sphinx-doc.org/en/master/>`_.
You can use the repository as a basis to build your own documentation system.
This is where you'll have to get a bit creative, but here's a rough outline
of the steps involved:
```

1. Add `godot-docs <https://hosted.weblate.org/projects/godot-engine/godot-docs>`_ as a submodule to your ``docs/`` folder.
2. Copy over its ``conf.py``, ``index.rst``, ``.readthedocs.yaml`` files into ``/docs/``. You may later decide to copy over more of godot-docs' files into your own project, like ``_templates/layout.html``.

> 3. Modify these files according to your project. This mostly involves adjusting paths to point to the ``godot-docs`` subfolder, as well as strings to reflect it's your project rather than godot you're building the docs for.

Reviewer: ```suggestion
3. Modify these files according to your project. This mostly involves adjusting paths to point to the ``godot-docs`` subfolder, as well as strings to reflect it's your project rather than Godot you're building the docs for.
```

4. Create an account on `readthedocs.org <http://readthedocs.org>`_. Import your project, and modify its base ``.readthedocs.yaml`` file path to ``/docs/.readthedocs.yaml``.

> Once you have completed all these steps, your documentation should be available at <repo-name>.readthedocs.io.

Reviewer: ```suggestion
Once you have completed all these steps, your documentation should be available at ``<repo-name>.readthedocs.io``.
```

        ``#ff00ff``, see :ref:`doc_bbcode_in_richtextlabel_hex_colors`).

    - ``[color={code/name}]{text}[/color]``

Publishing documentation online
-------------------------------

> You may want to publish an online reference for your GDExtension, akin to this website. The most important step is to build reStructuredText (``.rst``) files from your XML class reference:

Reviewer: ```suggestion
You may want to publish an online reference for your GDExtension, akin to this website.
The most important step is to build reStructuredText (``.rst``) files from your XML
class reference:
```

        ``#ff00ff``, see :ref:`doc_bbcode_in_richtextlabel_hex_colors`).

    - ``[color={code/name}]{text}[/color]``

Publishing documentation online
-------------------------------

> You may want to publish an online reference for your GDExtension, akin to this website.

Reviewer 2: ```suggestion
You may want to publish an online reference for your GDExtension, similar to this website.
```

Perhaps easier to understand than akin.

> The most important step is to build reStructuredText (``.rst``) files from your XML
> class reference:

Reviewer 2: ```suggestion
The most important step is to build reStructuredText (``.rst``) files from your XML class reference:
```

.. code-block:: none

    # You'll need to have Python installed for this command to work.
    curl -sSL https://raw.githubusercontent.com/godotengine/godot/master/doc/tools/make_rst.py | python3 - -o "docs/classes" -l "en" doc_classes

Your ``.rst`` files will now be available in ``docs/classes/``. From here, you can use
any documentation builder that supports reStructuredText syntax to create a website from them.
`godot-docs <https://hosted.weblate.org/projects/godot-engine/godot-docs>`_ uses `Sphinx <https://www.sphinx-doc.org/en/master/>`_.
You can use the repository as a basis to build your own documentation system.
This is where you'll have to get a bit creative, but here's a rough outline

> of the steps involved:

Reviewer 2: I think this should be phrased differently. Many of the people reading the docs will try to mimic the instructions even with a different setup so this should be phrased that the setup should be identical so the steps work. Otherwise we will get open issues complaining that the steps don't work.

Author: I'm trying to convey that the steps are very 'unprecise' with this sentence, because I needed to use a good amount of insight beyond this simple plan to actually get it to run. Describing all of the steps here would result in a pretty large and probably unhelpful and soon to be outdated tutorial. 

The only assumption the text is making is that the documentation files are in `doc_classes` (ensured by `doctool`). It should really work with any kind of setup beyond that.

Reviewer 2: @Ivorforce Perhaps along the lines of: 
> The setup may differ depending on how your project is structured. Here are the exemplary steps you would need to take to compile the documentation into rst files.

Author: I think i see where you're coming from, but the following steps actually do not depend on the project structure at all. This is just a guideline for how to adapt the Godot docs with different content. It's basically independent of the project itself, at least after the XML Generation step.

Author: I have edited the section with a new phrasing. Hopefully this conveys it better?
